### ISSUE TITLE
Recommended way to output logs in AsyncApp

### ISSUE THREAD

**Issue by tobenary** [slackapi/bolt-python#219]
Hi all, 

I know it might sound silly. But I do need some help regarding how to log users "requests" into a file with an async mode. 

Could you please spend a moment to describe the process for me?

**Comment by mwbrooks** [comment]
Hey @tobenary 👋🏻 this is definitely **not** a silly question. Off the top of my head, I don't have an answer for asynchronous logging but let me look into it for you!

**Comment by mwbrooks** [comment]
@tobenary You may already know a lot of this content, but I'll include it anyways to share my own discovery and learning process.

**TL;Dr;**

The main takeaway is that we'd recommend using the built-in logger module and setting it to write to a file. This aligns with the [official `asyncio` documentation's suggest](https://docs.python.org/3/library/asyncio-dev.html#asyncio-logger) and would work reliably with Bolt. When `asyncio` supports async logging, we'd be super happy to add support for it as well!

Below are a few of my learnings and research into `aiologger`:

**1. Writing logs to a file:**

[Bolt uses Python's standard logging module](https://slack.dev/bolt-python/concepts#logging), so you can [write logs to a file](https://docs.python.org/3/howto/logging.html#logging-to-a-file) using Python's standard approach:

```python
import logging
logging.basicConfig(filename='example.log', encoding='utf-8', level=logging.DEBUG)
```

**2. Python's logger is sync-only:**

Unfortunately, Python's logger is I/O Blocking. Even when using Bolt's Async methods based on `asyncio`, the logger will block with sync calls. This of course could be quite slow if you're writing to a file.

Interestingly, the [official `asyncio` documentation suggests using the built-in Python sync logger module](https://docs.python.org/3/library/asyncio-dev.html#asyncio-logger).

**3. Logging with middleware:**

You can use Bolt's middleware to log user requests. This would allow you to use a library such as [`aiologger`](https://b2wdigital.github.io/aiologger/usage.html) to asynchronous log the requests.

However, this middleware will not capture Bolt's internal logs which will still use the built-in Python logger.

**4. Using `aiologger` inside Bolt:**

Bolt allows you to [pass in your own `Logger` instance](https://github.com/slackapi/bolt-python/blob/v1.2.3/slack_bolt/app/async_app.py#L85). Since [`aiologger` uses the standard logging syntax](https://b2wdigital.github.io/aiologger/usage.html#migrating-from-standard-lib-logging), you could pass it into Bolt.

```python
import logging
import os
from aiologger import Logger
from slack_bolt.async_app import AsyncApp

# Workaround: must use log level > INFO for now
logger = Logger.with_default_handlers(level=logging.WARNING)

# Initializes your app with your bot token and signing secret
app = AsyncApp(
    token=os.environ.get("SLACK_BOT_TOKEN"),
    signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
    logger=logger,
)
```
However, I've came across some limitations:
1. [`.start()` is a sync method](https://github.com/slackapi/bolt-python/blob/main/slack_bolt/app/async_server.py#L72-L80), so Bolt will fail when using `aiologger` with a [log level above `INFO`](https://docs.python.org/3/library/logging.html#logging-levels). **We can discuss fixing this in a patch release if this is a method you'd like to use.**
2. It appears that the `aiologger` instance isn't passed down to the `slack_sdk` client. The result is that `slack_sdk` logs would not use your async logger that writes to a file. **This is unexpected behaviour, but we'd need to look into whether it's something that we can improve.**

**Comment by seratch** [comment]
Hi @tobenary, was the above comment helpful to you? As we've already provided an answer, let me close this issue after waiting for your response for a few more days.

**Comment by tobenary** [comment]
> We can discuss fixing this in a patch release if this is a method you'd like to use.
@seratch , @mwbrooks , What is my current options here?
1. Wait for @seratch fix in 10 days ( just writing a number )?
2. Have a workaround / patch as you suggested?

**Comment by seratch** [comment]
@tobenary 

>What is my current options here?

The short answer is that you can use the standard logging module as the asyncio document suggests. We do not recommend using aiologger or similar solutions along with Bolt at this point.

>Wait for @seratch fix in 10 days ( just writing a number )?
>Have a workaround / patch as you suggested?

No, we are not planning to apply any changes in months or probably years. 

> When asyncio supports async logging, we'd be super happy to add support for it as well!

As @mwbrooks mentioned the possibility of updates in the future, if the asyncio/logging module comes up with some solutions for this, we may be able to apply some changes. Nevertheless, as far as I know, we don't have any information about the probability.

>.start() is a sync method, so Bolt will fail when using aiologger with a log level above INFO. We can discuss fixing this in a patch release if this is a method you'd like to use.

Regarding this possibility of this patch, I'm not planning to apply anything in the short term. Indeed, it's possible to resolve the specific error in the `start()` method. However, this does not mean we can enable aiologger to properly work for all loggings.
